src/MySqlConnector/ColumnReaders/ColumnReaderFactory.cs

```diff
@@ -0,0 +1,204 @@
+namespace MySqlConnector.ColumnReaders;
+using MySqlConnector.Protocol;
+using MySqlConnector.Protocol.Payloads;
+using MySqlConnector.Protocol.Serialization;
+
+internal class ColumnReaderFactory
+{
+	internal static IColumnReader GetReader(bool binary, ColumnDefinitionPayload columnDefinition, MySqlConnection connection)
+	{
+		var isUnsigned = (columnDefinition.ColumnFlags & ColumnFlags.Unsigned) != 0;
+		if (binary)
+		{
+			switch (columnDefinition.ColumnType)
```

Comment:
There's quite a bit of duplicated logic between the `binary` and `!binary` `switch` statements that could be consolidated.

Will fix in https://github.com/mysql-net/MySqlConnector/pull/1332/commits/808f91fc42400b4bd88f75e3b7df6a9e590721d9.